src/ConfigurationClientManager.ts

```diff
@@ -0,0 +1,296 @@
+            this.endpoint = endpoint.toString();
+            this.#credential = credential;
+        } else {
+            throw new Error("A connection string or an endpoint with credential must be specified to create a client.");
+        }
+
+        this.#staticClients = [new ConfigurationClientWrapper(this.endpoint, staticClient)];
+        this.#validDomain = getValidDomain(this.endpoint);
+    }
+
+    async init() {
+        if (this.#appConfigOptions?.replicaDiscoveryEnabled === false || isBrowser() || isWebWorker()) {
+            this.isFailoverable = false;
+            return;
+        }
+
+        try {
+            this.dns = await import("dns/promises");
+        }catch (error) {
+            this.isFailoverable = false;
+            console.warn("Failed to load the dns module:", error.message);
+            return;
+        }
+
+        this.isFailoverable = true;
+    }
+
+    async getClients() : Promise<ConfigurationClientWrapper[]> {
+        if (!this.isFailoverable) {
+            return this.#staticClients;
+        }
+
+        const currentTime = Date.now();
+        if (this.#isFallbackClientDiscoveryDue(currentTime)) {
+            this.#lastFallbackClientRefreshAttempt = currentTime;
+            const host = new URL(this.endpoint).hostname;
+            await this.#discoverFallbackClients(host);
+        }
+
+        // Filter static clients whose backoff time has ended
+        let availableClients = this.#staticClients.filter(client => client.backoffEndTime <= currentTime);
+        // If there are dynamic clients, filter and concatenate them
+        if (this.#dynamicClients && this.#dynamicClients.length > 0) {
+            availableClients = availableClients.concat(
+                this.#dynamicClients
+                    .filter(client => client.backoffEndTime <= currentTime));
+        }
+
+        return availableClients;
+    }
+
+    async refreshClients() {
+        const currentTime = Date.now();
+        if (this.isFailoverable &&
+            currentTime > new Date(this.#lastFallbackClientRefreshAttempt + MINIMAL_CLIENT_REFRESH_INTERVAL).getTime()) {
+            this.#lastFallbackClientRefreshAttempt = currentTime;
+            const host = new URL(this.endpoint).hostname;
+            await this.#discoverFallbackClients(host);
+        }
+    }
+
+    async #discoverFallbackClients(host) {
+        let result;
+        try {
+            result = await Promise.race([
+                new Promise((_, reject) => setTimeout(() => reject(new Error("SRV record query timed out.")), SRV_QUERY_TIMEOUT)),
+                this.#querySrvTargetHost(host)
+            ]);
+        } catch (error) {
+            throw new Error(`Fail to build fallback clients, ${error.message}`);
+        }
+
+        const srvTargetHosts = result as string[];
+        // Shuffle the list of SRV target hosts
```

Comment:
Different from the .net provider's [implementation](https://github.com/Azure/AppConfiguration-DotnetProvider/blob/main/src/Microsoft.Extensions.Configuration.AzureAppConfiguration/ConfigurationClientManager.cs#L239), missed the sorting based on weight and priority.

@RichardChen820 Can you help double check this?

Reply:
Unresolved.